param/coercer/coercer.go

```diff
@@ -122,40 +101,78 @@ func coercePrimitiveType(val interface{}, primitiveType string) (interface{}, bo
 			return nil, false
 		}
 		return valFloat, true
+
+	case primitiveType == stringType:
```

Comment:
Added check for string primitive because in `anyOf` the first one is chosen. For example when given ["123", "456"], it should not be coerced to [123, 456] if string type is explicitly given. 
See test ``anyOf` with array of different primitive`